cinema/views.py

```diff
@@ -1,10 +1,21 @@
 from datetime import datetime
 
 from django.db.models import F, Count
-from rest_framework import viewsets
+from rest_framework import viewsets, mixins
+from rest_framework.authentication import TokenAuthentication
 from rest_framework.pagination import PageNumberPagination
+from rest_framework.permissions import IsAuthenticated
 
-from cinema.models import Genre, Actor, CinemaHall, Movie, MovieSession, Order
+from user.permissions import IsAdminOrIfAuthenticatedReadOnly
```

Comment:
The `IsAdminOrIfAuthenticatedReadOnly` permission is used for most viewsets. Verify that this permission aligns with the intended access control for these resources. If different permissions are needed, adjust accordingly.